### PR TITLE
Adding Smaller Alpine based image

### PR DESCRIPTION
**Pull request by saumitrabhave**
One major driving factor for creating this PR (Apart from smaller image size) is the version of postfix.
Alpine 3.8 has the postfix 3.3.x available in its package repo. while the ubuntu based image has 2.x AFAICT

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM tozd/runit
+FROM tozd/runit:ubuntu-bionic
```
Comment (mitar) [inline Dockerfile]: The `latest` is in fact equivalent to `ubuntu-trusty`. This is a backwards compatible issue. We should not be using that tag at all, but there it is. As you see, we do not yet use the explicit label everywhere.

So if you have not tested the image after changing this, change it to `ubuntu-trusty`.

Also, feel free to rename this file then to `ubuntu-trusty.dockerfile`.

File: etc/service/rsyslog/run [hunk 2]
```diff
@@ -6,4 +6,4 @@ rm -f /var/run/rsyslogd.pid
 mkdir -p /var/log/postfix
 chown syslog:syslog /var/log/postfix
 
-exec /usr/sbin/rsyslogd -n -c5 2>&1
```
Comment (mitar) [inline etc/service/rsyslog/run]: Why this change?

File: alpine-38.dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+VOLUME /var/log/postfix
+VOLUME /var/spool/postfix
+
+ENV MAILNAME mail.example.com
+ENV MY_NETWORKS 172.17.0.0/16 127.0.0.0/8
+ENV MY_DESTINATION localhost.localdomain, localhost
+ENV ROOT_ALIAS admin@example.com
+
+RUN apk update && apk add postfix rsyslog sed
```
Comment (mitar) [inline alpine-38.dockerfile]: Please keep the order and style of files. Then it is easier to compare them.

File: etc/service/rsyslog/run (hunk 2 above)
Comment (saumitrabhave) [reply etc/service/rsyslog/run]: `-c` parameter has been removed from rsyslogd ... on ubuntu based image it shows warning that this parameter will be ignored. But the alpine based image treats it as error, probably because it has much newer versions of the binaries.

File: etc/service/postfix/run [hunk 4]
```diff
@@ -24,8 +24,16 @@ fi
 
 # We have to start and stop postfix first through init.d to populate
 # postfix spool directory for chroot in which postfix is running.
+# We do it a bit differently in alpine as the init.d scripts use
+# openrc-run which is not supported by alpine very well, as the 
+# use on init systems is discouraged in general
+if [ -f /etc/alpine-release ]; then
+/usr/sbin/postfix -c /etc/postfix start >/dev/null 2>&1
```
Comment (mitar) [inline etc/service/postfix/run]: Maybe indent this for two spaces.

File: etc/service/rsyslog/run (hunk 2 above)
Comment (mitar) [reply etc/service/rsyslog/run]: Sounds good.

File: alpine-38.dockerfile (hunk 3 above)
Comment (mitar) [reply alpine-38.dockerfile]: Hm, frankly, those images are build automatically and often from scratch because the base image changed. So this optimization happens only if you are changing this manually, which is rare and probably only during development.

Anyway, `/etc/aliases` has to be available before postfix installation.

And please put every command into its own line.

Also, you can then modify the other dockerfile if you really want to use this order, but make them be the same.

Comment (saumitrabhave) [reply alpine-38.dockerfile]: Made changes to 1. keep each command on its own line. 2. move /etc/aliases before package installations. 3. keep both the files in similar manner in terms of sequence of execution

**Comment by mitar** [comment]
Ah, I remembered what the order is like that. Because you first have to configure installer using `debconf-set-selections` before installing on Ubuntu. With manual installation those would be prompts, but here you have to configure this in advance. So all `debconf-set-selections` lines have to come before installing `postfix`.

BTW, is there any configuration like that which has to be done on Alpine? So you should make sure the image you get is configured so that it is ready to deliver e-mail.

**Comment by saumitrabhave** [comment]
Alpine, has no prompts, as its main focus is containers, the default behaviour is to do things non-interactively. I test by baking the image and sending an actual email ...via the alpine based image.

**Comment by saumitrabhave** [comment]
Hi Can you please merge this, if there are no more suggessions.

**Comment by mitar** [comment]
So as I mentioned, in Debian reordering is not a good because prompts have to be first configured. I think I would prefer now the previous order of commands, and that both have the same order.

**Comment by saumitrabhave** [comment]
Hi @mitar , I closed this PR, because my use cases started varying a bit more, like I need pcre and mysql support in the image, and I assume more will be needed (Like Immutable images and versioning, to make sure that I wont modify the same tag again and again)

Please, note the deviations in postfix versions and plugins that have been merged in alpine image.

Appreciate the great work you have done already, helped me a lot.
